BUILDING.md

```diff
@@ -155,3 +179,16 @@ installed to be able to compile the C/C++ sources.
 > cmake .. -GNinja -DCMAKE_BUILD_TYPE=Release
 > cmake --build .
 ```
+
+If you want to build DLLs, run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
+
+## cmake: Install After Build
```

Comment:
the main documentation in terms of how to use gRPC as a user is here: https://github.com/grpc/grpc/blob/master/src/cpp/README.md, so perhaps we should move there? (and only add a note here?).
Btw, this is already mentioned here: https://github.com/grpc/grpc/blob/master/BUILDING.md#build-from-source (feel free to rephrase if the wording is not clear).

Reply:
The README.md section about [make](https://github.com/grpc/grpc/blob/master/src/cpp/README.md#make) points back to BUILDING.md. It seems weird to me to have part of the build doc be in README.md and part be in BUILDING.md. We should make it consistent.

Reply:
The idea was that the section about `make` in README.md points back to BUILDING.md for only for the build instructions, and includes the extra instructions on how to install and I expected we would follow the pattern for cmake as well.

I agree with the point that perhaps having part of the build instructions in one document and the other part in another document, but I wanted to make a very clear difference between 
1. contributor-only instructions for building and developing gRPC itself
2. instructions for including gRPC C++ as a dependency in another project (as a user).

Ideally the build part of the instructions for the users should be so simple, that duplicating it in README.md shouldn't be a problem and the link to BUILDING.md  would be there only to troubleshooting purposes or as a way to learn more about the build. WDYT?

Reply:
I'm happy to consider alternatives - feel free to propose how BUILDING.md and README.md should be restructured (a general proposal is fine).

Btw, over time we will be deprecating `make` for installation and for building (make as in "standalone make" not as a generator in cmake), so we'll probably want to make that clear in the docs (and reorder the build systems by their importance).